discovery-provider/src/utils/helpers.py

```diff
@@ -96,6 +110,52 @@ def configure_logging(loglevel_str='WARN'):
     logger.setLevel(loglevel)
 
 
+def configure_flask_app_logging(app, loglevel_str):
+    # Disable werkzeug logger to remove default flask logs in favor of app decorator log below
+    werkzeug_log = logging.getLogger('werkzeug')
+    werkzeug_log.disabled = True
+
+    configure_logging(loglevel_str)
+    logger = logging.getLogger()  # retrieve root logger
+
+    # Set a timer before the req to get the request time
+    @app.before_request
+    def start_timer(): # pylint: disable=W0612
+        g.start = time.time()
+
+    # Log the request
+    @app.after_request
+    def log_request(response): # pylint: disable=W0612
+
+        now = time.time()
+        duration = int((now - g.start) * 1000)
+        ip = request.headers.get('X-Forwarded-For', request.remote_addr)
+        host = request.host.split(':', 1)[0]
+        args = request.query_string.decode("utf-8")
+
+        log_params = {
+            'method': request.method,
+            'path': request.path,
+            'status': response.status_code,
+            'duration': duration,
+            'ip': ip,
+            'host': host,
+            'params': args
+        }
+
+        request_user_id = request.headers.get('X-User-ID')
+        if request_user_id:
+            log_params['request_user_id'] = request_user_id
+
+        parts = []
+        for name, value in log_params.items():
+            part = "{}={}".format(name, value)
+            parts.append(part)
+
+        logger.info('handle flask request', extra=log_params)
```

Comment:
so this log is output with every request now?

Reply:
Yeah, each request would log something like this.
<img width="737" alt="Screen Shot 2020-08-17 at 11 23 03 AM" src="https://user-images.githubusercontent.com/7064122/90413163-1c162000-e07c-11ea-8e80-86c1f0145d0c.png">

Reply:
ah makes sense, cool